src/Node.ts

```diff
@@ -915,8 +915,8 @@ export abstract class Node<Config extends NodeConfig = NodeConfig> {
    * @name Konva.Node#getAttrs
    * @returns {Object}
    */
-  getAttrs() {
-    return (this.attrs || {}) as Config & Record<string, any>;
+  getAttrs(): Partial<Config> {
```

Comment:
I'd like to type `this.attrs` as `Partial<Config>`, but this will require some minor code changes, so I've left it out of this PR.

Reply:
The issue with this type, as it limits to config only. But it is possible to write ANY custom data into attributes.